### DISCUSSION TITLE
如何安装第三方插件的依赖

### DISCUSSION THREAD

**Discussion by ilyfairy** [nonebot/discussions#488] [问答]
文件夹结构:
```
/pyproject.toml
/plugins/插件1/pyproject.toml
/plugins/插件1/插件1/__init__.py
```

我指定`plugin_dirs = ["plugins"]`不起作用, 要指定`plugin_dirs = ["plugins/插件1"]`才行, 但是它不自动安装依赖, 莫非我要手动找到`[tool.poetry.dependencies]`的依赖一个个安装

如果我通过`nb plugin`安装插件, 它会安装到.venv里面, 而不是plugins文件夹里  
我要安装并修改这个第三方插件的源码, 所以用`nb plugin`不太合适

是我操作的方式有问题么, 要怎么弄?

**Comment by yanyongyu** [comment]
如果你是把这个插件项目整个下载下来的，你应该通过 `plugins=["plugins.插件1.插件1"]` 来加载，不推荐使用 dir 方式。或者你可以把他仓库里的插件部分单独拿出来放在 plugins 目录下加载。

**Comment by ilyfairy** [comment]
通过`plugins=["plugins.插件1.插件1"]`加载, 它还是提示模块找不到, `plugins.插件1`还依赖其它模块, 定义在`插件1`的`pyproject.toml`.`[tool.poetry.dependencies]`中  
没有类似`npm install`的方法来自动安装这些依赖么

**Reply by yanyongyu** [reply]
你可以在项目依赖里添加这个文件夹为依赖，请参考 poetry 文档
